Breeding-Insight/bi-web
# BI-1910 - Allow Breeder to Upload a Genotype Sample File

# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1910

Created a new UI component named "Sample Management"

- Defined a new import type: Genotype Sample
- Defined a new import template
- Created a list view of all sample submissions
- Created a detailed view of an individual submission
- Implemented ability to manual update the status of a submission 
- Implemented the ability to automatically submit to a vendor and check the status


# Dependencies
- bi-api/feature/BI-1910
- bi-docker-stack/feature/BI-1910

# Testing

## Testing with germplasm
1. Create germplasm records in your program
2. On the Import Data page, go to the "Genotype Samples" tab, and download the template
3. Populate the template, and make sure to populate the GID column
4. Upload the sample submission file, and ensure the preview is correct, and confirming succeeds
5. Navigate to the submission's detail page, and ensure the GID column is populated

## Testing with ObsUnitIDs
1. Create germplasm records in your program
2. Create an experiment
3. On the Import Data page, go to the "Genotype Samples" tab, and download the template
4. Populate the template, and make sure to populate the ObsUnitID column
6. Upload the sample submission file, and ensure the preview is correct, and confirming succeeds
7. Navigate to the submission's detail page, and ensure the ObsUnitID and GID columns are populated

## Manual Submission
1. When on the submission detail page, and the submission has not been submitted, click on "Manage Submission" and you should see an option for "Manual Update Submission"
2. Clicking on "Manual Update Submission" should open a modal window with a dropdown allowing you to change the submissions status
3. Change the status to "SUBMITTED" and click save, ensure save succeeds and the metadata on the detail page is updated

## Automated Submission
1. Set the `BRAPI_VENDOR_SUBMISSION_ENABLED` property to `true` for both bi-web and bi-api and restart
2. When on the submission detail page, and the submission has not been submitted, click on "Manage Submission" and you should see an option for "Submit to DArT"
3. Clicking on "Submit to DArT" should open a modal window asking you to confirm this action
4. Click "Submit"
5. Ensure the submission was successful and that the metadata on the detail page is updated to show the vendor submission ID and status
6. Click on "Manage Submission" and you should see an option for "Check Vendor Status".
7. Click "Check Vendor Status", and a modal should appear with a loading indicator
8. Ensure there are no errors, and that the "Last Status Check" is updated to the current time

## General testing
1. Ensure that all successful uploads display in the Sample Management table, and show correct status information and metadata
9. Ensure when viewing a submission detail that the metadata is correct, the table containing the sample information is correct, and the tab containing plate layouts is correct



# Checklist:
